analyze/edge-light.ts

```diff
@@ -161,3 +187,56 @@ export async function detectBot(
     };
   }
 }
+export async function detectSensitiveInfo<Custom extends string>(
+  context: AnalyzeContext,
+  candidate: string,
+  options?: SensitiveInfoConfig<Custom>,
+): Promise<DetectSensitiveInfoResult<Custom>> {
+  const analyze = await init(context, options?.customDetect);
+  const skipCustomDetect = options?.customDetect === undefined;
+  let entities: core.Entities = {
+    tag: "allow",
+    val: options?.allow?.map(ConvertProtocolEntitiesToAnalyzeEntities) || [],
+  };
+  if (options?.deny) {
+    entities = {
+      tag: "deny",
+      val: options.deny.map(ConvertProtocolEntitiesToAnalyzeEntities),
+    };
+  }
+  const optionsOrDefault = {
+    entities,
+    contextWindowSize: options?.contextWindowSize || 1,
+    skipCustomDetect,
+  };
+
+  if (typeof analyze !== "undefined") {
+    const result = analyze.detectSensitiveInfo(candidate, optionsOrDefault);
+    const allowed = result.allowed.map((entity) => {
+      return {
+        ...entity,
+        identifiedType: ConvertAnalyzeEntitiesToProtocolEntities(
+          entity.identifiedType,
+        ),
+      };
+    });
+    const denied = result.denied.map((entity) => {
+      return {
+        ...entity,
+        identifiedType: ConvertAnalyzeEntitiesToProtocolEntities(
+          entity.identifiedType,
+        ),
+      };
+    });
+    return {
+      allowed,
+      denied,
+    };
+  } else {
+    // Skip the local evaluation of the rule if WASM is not available
+    return {
+      allowed: [],
+      denied: [],
+    };
```

Comment:
I'm wondering if this is really bad for a local-only rule. Perhaps we want to throw?